cmd/omniWatcher.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2018 Vulcanize
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/spf13/cobra"
+
+	"github.com/vulcanize/vulcanizedb/libraries/shared"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+	"github.com/vulcanize/vulcanizedb/pkg/geth"
+	"github.com/vulcanize/vulcanizedb/pkg/geth/client"
+	vRpc "github.com/vulcanize/vulcanizedb/pkg/geth/converters/rpc"
+	"github.com/vulcanize/vulcanizedb/pkg/geth/node"
+	"github.com/vulcanize/vulcanizedb/pkg/omni/transformer"
+	"github.com/vulcanize/vulcanizedb/pkg/omni/types"
+)
+
+// omniWatcherCmd represents the omniWatcher command
+var omniWatcherCmd = &cobra.Command{
+	Use:   "omniWatcher",
+	Short: "Watches events at the provided contract address",
+	Long: `Uses input contract address and event filters to watch events
+
+Expects an ethereum node to be running
+Expects an archival node synced into vulcanizeDB
+Requires a .toml config file:
+
+  [database]
+  name = "vulcanize_public"
+  hostname = "localhost"
+  port = 5432
+
+  [client]
+  ipcPath = "/Users/user/Library/Ethereum/geth.ipc"
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		omniWatcher()
+	},
+}
+
+func omniWatcher() {
+	if contractAddress == "" && len(contractAddresses) == 0 {
+		log.Fatal("Contract address required")
+	}
+
+	if !methodsOn && !eventsOn {
+		log.Fatal("Method polling and event watching turned off- nothing to do!")
+	}
+
+	if len(contractEvents) == 0 || len(contractMethods) == 0 {
+		var str string
+		for str != "y" {
+			reader := bufio.NewReader(os.Stdin)
+			if len(contractEvents) == 0 && len(contractMethods) == 0 {
+				fmt.Print("Warning: no events or methods specified, proceed to watch every event and method? (Y/n)\n> ")
+			} else if len(contractEvents) == 0 {
+				fmt.Print("Warning: no events specified, proceed to watch every event? (Y/n)\n> ")
+			} else {
+				fmt.Print("Warning: no methods specified, proceed to watch every method? (Y/n)\n> ")
+			}
+			resp, err := reader.ReadBytes('\n')
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			str = strings.ToLower(string(resp))
+			if str == "n" {
```

Comment:
Pretty minor, but I'm wondering if we might want want to invert this so that we require an affirmative response to continue (e.g. `if str[0] != "y"`), rather than proceeding unless they enter `"n"`/`"N"`

Reply:
I believe as it is it should only proceed if they enter Y or y, will exit if they enter N or n, and will continue to loop and wait for input if they enter anything else. Using str[0] would be better.

Reply:
oh nice, missed the `for str != "y"` part 🤦‍♂️